### ISSUE TITLE
`prefer-const` requires all destructured array items to be the same

### ISSUE THREAD

**Issue by intellix** [palantir/tslint#2636]
### Bug Report

- __TSLint version__: 5.1.0
- __TypeScript version__: 2.2.2
- __Running TSLint via__: (pick one) CLI / VSCode

#### TypeScript code being linted

```ts
let [month, year] = '12/12'.split('/').map(v => +v);
year += 2000;
```

with `tslint.json` configuration:

```json
"prefer-const": true,
```

#### Actual behavior
[tslint] Identifier 'month' is never reassigned; use 'const' instead of 'let'. (prefer-const)

#### Expected behavior
No error. I can't use const because I'm altering one of the vars so both need to be `let` in such a case.

**Comment by ajafff** [comment]
that's what you want:
```js
"prefer-const": [true, {"destructuring": "all"}],
```

**Comment by intellix** [comment]
Damn, searched issues but couldn't find anything. Should have checked the docs. Thanks :)
